### PR TITLE
Implement watching for filesystem storage

### PR DESCRIPTION
**Pull request by vladklokun**
# What this PR changes?

This PR implements the Watch method for the filesystem storage implementation.

### CODE REVIEW

File: pkg/registry/file/watch.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+type watchDispatcher struct {
+	watchesByKey *xsync.MapOf[string, watchersList]
+}
+
+func newWatchDispatcher() watchDispatcher {
+	wbk := xsync.NewMapOf[watchersList]()
+	return watchDispatcher{watchesByKey: wbk}
+}
+
+func exractKeysToNotify(key string) ([]string, error) {
```
Comment (matthyx) [inline pkg/registry/file/watch.go:63]: nit: extract

**Review by matthyx — APPROVED** [review]
let's rebase and merge this (maybe check the small nit)
we can fix the cleanup later

**Comment by matthyx** [comment]
FYI I have merged https://github.com/kubescape/storage/pull/20

File: pkg/registry/file/watch.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+type watchersList []*watcher
+
+// watchDispatcher dispatches events to registered watches
+//
+// TODO(vladklokun): Please keep in mind that this dispatcher does not offer any collection of
+// resources left over by the stopped watches! There are multiple ways to go about it:
+// 1. On-Stop cleanup, where a watcher would notifies the dispatcher about being stopped, and that it can be cleaned up
+// 2. Garbage collection. Periodic background cleanup. Poses challenges as this
+// might be a contended concurrent resource.
```
[the remark this answers is not in the document]
Comment (vladklokun) [reply pkg/registry/file/watch.go:53]: I agree.

**Review by matthyx — APPROVED** [review]

**Pull request merged**